examples/chaos.py

```diff
@@ -0,0 +1,95 @@
+
+"""
+This example demonstrates the Polynomial Choas Expansion (PCE) Surrogate Direct Model functionality. PCE is a method by which the behavior of a model can be approximated by a polynomial. In this case the relationship between future loading and time of event. The result is a direct surrogate model that can be used to estimate time of event given a loading profile, without requiring the original model to be simulated. The resulting estimation is MUCH faster than simulating the model.
+
+This functionality is especially useful in cases where you need to evaluate a large number of loading profiles in a short amount of time or with limited computational resources. For example, when you would like to optimize a loading profile. In that case, it would not make sense to have a state-transition model in the inner loop of the optimization, since the state transition model would be called many times. Instead, you can use a direct surrogate model to estimate time of event directly from the loading profile.
+
+In this example, a PCE surrogate model is generated for the BatteryElectroChemEOD model. The surrogate is used to estimate time of event for a number of loading profiles. The result is compared to the actual time of event for the same loading profiles, which were generated by simulating the model.
+"""
+
+import chaospy as cp
+import matplotlib.pyplot as plt
+import numpy as np
+from prog_models.models import BatteryElectroChemEOD
+from prog_models.data_models import PCE
+import scipy as sp
+
+def run_example():
+    # First lets define some constants
+
+    # Time step used in simulation
+    DT = 0.5 
+
+    # The number of samples to used in the PCE
+    # Larger gives a better approximation, but takes longer to generate
+    N_SAMPLES = 100  
+
+    # The distribution of the input current
+    # This defines the expected values for the input
+    # In this case we're saying that the input current can be anything between 3-8 amps
+    # With a uniform distribution (i.e., no value in that range is more likely than any other)
+    INPUT_CURRENT_DIST = cp.Uniform(3, 8)
+    # Note: These discharge rates are VERY high. This is only for demonstration purposes.
+    #    The high discharge rate will accelerate the degradation of the battery, 
+    #    which will cause the example to run faster
+
+    # Step 1: Define base model
+    # First let's define the base model that we're creating a surrogate for.
+    m = BatteryElectroChemEOD(process_noise = 0) 
+    x0 = m.initialize()  # Initial State
+    
+    # Step 2: Build surrogate
+    # Next we build the surrogate model from the base model
+    # To build the model we pass in the distributions of possible values for each input.
+    # We also provide the max_time. This is the maximum time that the surrogate will be used for.
+    # We dont expect any battery to last more than 4000 seconds given the high discharge curves we're passing in.
+    m_surrogate = PCE.from_model(m, 
```

Comment:
Just checking: Did you mean to not include a PCE order that is specified by the user? Including it might be too much for an introductory example - I'm not sure.